### ISSUE TITLE
Congfigurable product converts to a simple on save

### ISSUE THREAD

**Issue by christian-walker** [magento/magento2#10357]
### Preconditions

1. Magento version 2.1.7
2. PHP7 and Magemojo default MySQL version 5.6.33

### Steps to reproduce

1. Click into a configurable product
2. Change a few custom attribute settings 
3. Click Save

### Expected result

1.  Configurable product should save, stay as a configurable product and the simple products should stay associated.

### Actual result

1. Configurable is converted to a simple product, therefore the simples are unlinked.

**Comment by orlangur** [comment]
Please describe step 2 "Change a few custom attribute settings" in more details.

**Comment by christian-walker** [comment]
@orlangur sorry it's really any attribute. It could be the name, weight, or any additional attributes you create. It doesn't happen all the time either, which is the weird. Very inconsistent.

**Comment by orlangur** [comment]
I see, thanks! Just wanted to understand whether it's particular attribute type or something.

**Comment by brazillia** [comment]
@ChristianWalker19, did you come up with a solution for this? I'm having the same problem.

I'm just starting my migration from 1.9.3 to 2.1.8 and I haven't installed any extensions yet so it's a pretty clean installation. 

If I create my configurable product, auto-create the configurations, and view the product on the front end, everything looks good. I can change the name and sku of the configurations and the configurable product still displays correctly. But as soon as I make a single edit to the configurable product (e.g. adding or changing the description) the parent/child link is broken (though the parent still remains as a configurable product). 

I've looked through all the catalog_product tables in the database but the only change I noticed was in the catalog_product_relation table where the parent and child ID's were no longer linked. Just adding the entries back in, however, didn't restore the links.

Thanks for any suggestions.

**Comment by EmilyPepperman** [comment]
I have the same problem now... since we updated to 2.1.8 from 2.1.7 !!

**Comment by brazillia** [comment]
I just installed the latest 2.2 release candidate and it looks like it's fixed (at the moment). I was able to auto create the child products, change the product description without breaking the parent/child link, create a separate simple product and change the auto created child with my simple product. 

Now we just have to hope it's still working in the final release, and that the release date isn't too far away.
